frontend/src/stores/shopingCart.ts

```diff
@@ -0,0 +1,37 @@
+import { action, makeObservable, observable } from 'mobx';
+import { RootStore } from '.';
+import { Course, ShopingCart } from '../app-types';
+
+export class ShopingCartStore {
+  root: RootStore;
+
+  cartPayload: ShopingCart = {
+    course: [],
+  };
+
+  constructor(root: RootStore) {
+    this.root = root;
+    makeObservable(this, {
+      cartPayload: observable,
+      addToCart: action.bound,
+      deleteCourseFromCart: action.bound,
+    });
+  }
+
+  addToCart(course: Course) {
+    if (
+      this.cartPayload.course.find(
+        (courseInCart) => courseInCart.id === course.id,
+      )
+    ) {
+      return;
+    }
+    this.cartPayload.course.push(course);
+  }
```

Comment:
Myślę że można to prościej zapisać
```js
const foundCourse = this.cartPayload.course.find((courseInCart) => courseInCart.id === course.id)

if (!foundCourse) {
 this.cartPayload.course.push(course);
}

```

Sam kod o wiele czytelniejszy i bezniepotrzebnego returna :D